scripts/blog.js

```diff
@@ -13,43 +13,66 @@ blog.makePosts = function(){
 };
 
 
-
 blog.sortArticles = function () {
   blog.rawData.sort(function (a, b) {
-  if (a.publishedOn < b.publishedOn) {
-    return 1;
-  }
-  if (a.publishedOn > b.publishedOn) {
-    return -1;
-  }
-  return 0;
-});
-}
+    if (a.publishedOn < b.publishedOn) { return 1;}
+    if (a.publishedOn > b.publishedOn) { return -1;}
+    return 0;
+  });
+};
+
 
 blog.dateDiff = function(date1){
-  var today = new Date();
-  var dd = today.getDate();
-  var mm = today.getMonth()+1;
-  var yyyy = today.getFullYear();
+  return(Math.floor((new Date() - new Date(date1)) / 86400000));
+};
 
-  if(dd<10) {
-    dd='0'+dd;
-  }
+blog.truncateArticles = function () {
+  $('article section p:not(:first-child)').hide();
+  $('main').on('click', '.read-on', function(event){
+    event.preventDefault();
+    $(this).parent().find('p').fadeIn();
+    $(this).hide();
+  });
+};
 
-  if(mm<10) {
-    mm='0'+mm;
-  }
+blog.filterViewByAuthor = function () {
+  var options = $.map(blog.rawData, function(post, idx) {
+    var optionTag = $('<option>').val(post.author).text(post.author);
+    optionTag.data('rawdata', post);
+    return optionTag;
+  });
 
-  today = yyyy +'-'+ mm+'-'+dd;
+  $.fn.append.apply($('.author-filter'), options)
+    .change(function() {
+      $('article').show();
+      $('.category-filter').children().removeAttr('selected');
+      var $hoo = $('.author-filter option:selected').val();
+      $('article h5:not(:contains("'+ $hoo +'"))').parent().hide();
+
+    });
+};
 
```

Comment:
can you pick more descriptive variable names than hoo and boo (below)? I'm not sure what those are signifying, but it seems like it's the target attribute. remember it's ok to reuse variable names if they are contained in a function (not global).